tests/Integration/Behaviour/Features/Context/Domain/ContactFeatureContext.php

```diff
@@ -0,0 +1,145 @@
+    {
+        $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+    }
+
+    /**
+     * @When I add new contact with the following properties:
+     *
+     * @param TableNode $table
+     */
+    public function iAddNewContactWithTheFollowingProperties(TableNode $table)
+    {
+        $data = $this->extractFirstHorizontalRowFromProperties($table);
+        /** @var EditableContact $editablContact */
+        $editableContact = $this->mapToEditableContact(self::DUMMY_CONTACT_ID, $data);
+
+        $addContactCommand = new AddContactCommand(
+            $editableContact->getLocalisedTitles(), $editableContact->isMessagesSavingEnabled()
+        );
+        $addContactCommand->setEmail($editableContact->getEmail()->getValue());
+        $addContactCommand->setLocalisedDescription($editableContact->getLocalisedDescription());
+        $addContactCommand->setShopAssociation($editableContact->getShopAssociation());
+
+        $this->getCommandBus()->handle($addContactCommand);
+    }
+
+    /**
+     * @When contact with id :contactId should have the following properties:
+     *
+     * @param int $contactId
+     * @param TableNode $table
+     */
+    public function contactWithIdShouldHaveTheFollowingProperties(int $contactId, TableNode $table)
+    {
+        $data = $this->extractFirstHorizontalRowFromProperties($table);
+        $expectedEditableContact = $this->mapToEditableContact($contactId, $data);
+        /** @var EditableContact $editableContact */
+        $editableContact = $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+        PHPUnit_Framework_Assert::assertEquals($expectedEditableContact, $editableContact);
+    }
+
+    /**
+     * @When I update contact with id :contactId with the following properties:
+     *
+     * @param int $contactId
+     * @param TableNode $table
+     */
+    public function iUpdateContactWithIdWithTheFollowingProperties(int $contactId, TableNode $table)
+    {
+        $data = $this->extractFirstHorizontalRowFromProperties($table);
+        $editableContact = $this->mapToEditableContact($contactId, $data);
+
+        $editContactCommand = new EditContactCommand($contactId);
+        $editContactCommand->setLocalisedTitles($editableContact->getLocalisedTitles());
+        $editContactCommand->setShopAssociation($editableContact->getShopAssociation());
+        $editContactCommand->setLocalisedDescription($editableContact->getLocalisedDescription());
+        $editContactCommand->setEmail($editableContact->getEmail()->getValue());
+        $editContactCommand->setIsMessagesSavingEnabled($editableContact->isMessagesSavingEnabled());
+
+        $this->getCommandBus()->handle($editContactCommand);
+    }
+
+    /**
+     * @param int $contactId
+     * @param array $data
+     *
+     * @return EditableContact
+     */
+    private function mapToEditableContact(int $contactId, array $data): EditableContact
+    {
+        $title = $data['title'];
```

Comment:
You don't need this temporary variables you can use them in the constructor right away

Reply:
@jolelievre @matks @rokaszygmantas In this Behat case, yes, just thought about this style if it is not the Behat test code but production code and the error happens on the live system. Let's imagine the situation, Prestashop shop is running on heavy loaded system and every minute gives profit for the company. Error happens and developer need to fix it ASAP. The case is specific to the LIVE system and in this style of code it won't be obvious which code caused the error - only the long error line will be visible and debugging might be needed.

So not sure what gives more business value to PrestaShop:

1. Quickly fixing production errors by developers Or
2. Speed / memory usage of software
3. My assumptions might be not very good also, because I came from the company with limited knowledge of E-commerce and it was more important to fix errors in production environment ASAP. Successful production environment running time was traded for money and customers value; It might be that I don't understand something about live system maintenance - as this style was forced to me by another more experienced developer.